tests/dhcp_relay/test_dhcp_relay_stress.py

```diff
@@ -95,3 +98,123 @@ def _check_socket_buffer():
                             "uplink_mac": str(dut_dhcp_relay_data[0]['uplink_mac']),
                             "testing_mode": testing_mode},
                    log_file="/tmp/dhcp_relay_test.stress.DHCPTest.log", is_python3=True)
+
+
+def check_dhcp_stress_status(duthost, test_duration_seconds):
+    # Monitor DHCP status during the test
+    start_time = time.time()
+    sleep_time = 30
+    while time.time() - start_time < test_duration_seconds - sleep_time:
+        # Check the status of the DHCP container
+        dhcp_container_status = duthost.shell('docker ps | grep dhcp_relay')["stdout"]
+        if dhcp_container_status == "":
+            assert False, "DHCP container is NOT running."
+
+        # Check CPU usage of the DHCP process
+        dhcp_cpu_usage = duthost.shell('show processes cpu --verbose | grep dhc | awk \'{print $9}\'')["stdout"]
+        if dhcp_cpu_usage:
+            dhcp_cpu_usage_lines = dhcp_cpu_usage.splitlines()
+            for cpu_usage in dhcp_cpu_usage_lines:
+                cpu_usage_float = float(cpu_usage)
+            assert cpu_usage_float < 50.0, "DHCP CPU usage is too high: {}%".format(cpu_usage_float)
+
+        # Check the status of multiple DHCP processes inside the container
+        dhcp_process_status = duthost.shell(
+             'docker exec dhcp_relay supervisorctl status | grep dhcp | grep -v dhcp6')["stdout"]
+        if dhcp_process_status:
+            dhcp_process_status_lines = dhcp_process_status.splitlines()
+            for dhcp_process_status_line in dhcp_process_status_lines:
+                process_name, process_status = dhcp_process_status_line.split()[0], dhcp_process_status_line.split()[1],
+                assert process_status == "RUNNING", "{} is not running!".format(process_name)
+    time.sleep(sleep_time)
+
+
+@pytest.mark.parametrize('dhcp_type', ['discover', 'offer', 'request', 'ack'])
+def test_dhcp_relay_stress(ptfhost, ptfadapter, dut_dhcp_relay_data, validate_dut_routes_exist,
+                           testing_config, dhcp_type, clean_processes_after_stress_test):
+    """Test DHCP relay functionality on T0 topology
+       and verify that HCP relay service can handle the maximum load without failure.
+    """
+    testing_mode, duthost = testing_config
+    packets_send_duration = 120
+    client_packets_per_sec = 10000
+
+    if testing_mode == DUAL_TOR_MODE:
```

Comment:
I don't think we need this skip condition, this test case wouldn't be backported to 2018 or 2019 branch